urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/JuegoTorneo.java

```diff
@@ -0,0 +1,194 @@
+					hacerMovida();
+				} else {
+					juegoTerminado();
+				}
+			}	
+		}
+	}
+	
+	private Set<Point> marcarMovidasPosibles() {
+		Set<Point> movidas = controlador.marcarMovidasPosibles();
+		controlador.desmarcarMovidasPosibles();
+		return movidas;
+	}
+	
+	private void hacerMovida() {
+		/**///System.out.println("entró a jugar PC"+controlador.jugadorActual().color().toString());
+		Point movidaComputador = controlador.evaluarMovida();
+		controlador.hacerMovimiento(movidaComputador);
+		controlador.cambiarTurno();
+		run();
+	}
+	
+	@SuppressWarnings("deprecation")
+	private void juegoTerminado() {
+		/*if (!controlador.esEmpate()) {
+			colorJugadorGanador = controlador.obtenerGanador().color() == EstadoCasilla.BLACK ? "Negras" : "Blancas";
+			estrategiaGanador = controlador.obtenerGanador().getEstrategia();
+		}*/
+		this.stop();
+	}
+	
+
+	public int[] getGanador() {
+		if (!controlador.esEmpate()) {
+			/**///String colorJugadorGanador = controlador.obtenerGanador().color() == EstadoCasilla.BLACK ? "Negras" : "Blancas";
+			/**///System.out.println("Ganó "+colorJugadorGanador+"!!!");
+			return controlador.obtenerGanador().getEstrategia();
+		}
+		return null;
+	}
+	
+	public void definirTipoTablero(int tamanio, TipoTablero tipoTablero){	
+		String nombreFichero = "Octogonal10x10.oth";
+		File directorio = new File("setup/boards");
+		if ( tipoTablero == TipoTablero.OCTOGONAL ) {
+			JFileChooser fileChooser = new JFileChooser();
+			nombreFichero = "Octogonal10x10.oth";
+			if (tamanio == 12) {
+				nombreFichero = "Octogonal12x12.oth";
+			} else if (tamanio == 14) {
+				nombreFichero = "Octogonal14x14.oth";
+			} else if (tamanio == 16) {
+				nombreFichero = "Octogonal16x16.oth";
+			}
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			fileChooser.setCurrentDirectory(fichero);		
+			fileChooser.setFileFilter(miFiltro);
+			try {
+				leerArchivo(fichero, tipoTablero);
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			}
+		}
+		
+		if ( tipoTablero == TipoTablero.CLASICO || tipoTablero == null) {
+			controlador.inicializar(tamanio, tamanio, tipoTablero);
+		}
+		
+		if ( tipoTablero == TipoTablero.PERSONALIZADO ){
+			//controlador.inicializar(tamanio, tamanio, tipoTablero);
+			nombreFichero = "Personalizado.oth";
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			try {
+				leerArchivo(fichero, TipoTablero.OCTOGONAL);
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			}
+		}
+		//run();
+	}
+	
+	private void leerArchivo(final File fichero, TipoTablero tipoTablero) throws FileNotFoundException {
+		BufferedReader reader;
+		int largoTablero=14;
+		int anchoTablero=14;
+		Map<Point, EstadoCasilla> crearTablero = new HashMap<Point, EstadoCasilla>();
+		EstadoCasilla colorJugador = EstadoCasilla.BLACK;
+		boolean esTamańo=true;
+		
+		try {
+		    int casillasJugablesIniciales=0;
+			reader = new BufferedReader(new FileReader(fichero));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced at lines 158, 159, 180. The resource is closed at line 186. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).